cpp/src/arrow/type_traits.h

```diff
@@ -430,6 +430,17 @@ static inline bool is_binary_like(Type::type type_id) {
   return false;
 }
 
+static inline bool is_dictionary(Type::type type_id) {
+  switch (type_id) {
```

Comment:
Any reason not to simplify this to `return type_id == Type::DICTIONARY`?

Reply:
nope, fixing